clang/lib/CIR/CodeGen/CIRGenExprComplex.cpp

```diff
@@ -82,6 +82,17 @@ class ComplexExprEmitter : public StmtVisitor<ComplexExprEmitter, mlir::Value> {
   }
 
   mlir::Value VisitUnaryDeref(const Expr *e);
+
+  mlir::Value VisitUnaryPlus(const UnaryOperator *e,
+                             QualType promotionType = QualType());
```

Comment:
I don't think `VisitUnaryPlus` and `VisitUnaryMinus` are ever called with a `promotionType` argument. I looked at the commit where these arguments were added in classic codegen (https://github.com/llvm/llvm-project/commit/5def954a5b1774747bbe90a4f6879e51f341a74d) and it looks like it was possibly an artifact of some previous state. I'd suggest leaving them out for now.